test/rubygems/helper.rb

```diff
@@ -1491,6 +1491,23 @@ def self.key_path(key_name)
     "#{__dir__}/#{key_name}_key.pem"
   end
 
+  ##
+  # Sets default browser to Safari in macOS by saving current configuration and overwriting LSHandlers
+
+  def set_default_browser_safari
+    @launch_services = `defaults read com.apple.LaunchServices/com.apple.launchservices.secure LSHandlers`
+    default_browser_setting = "{ LSHandlerURLScheme = \"https\"; LSHandlerRoleAll = \"com.apple.safari\"; }"
+
+    `defaults write com.apple.LaunchServices/com.apple.launchservices.secure LSHandlers -array '#{default_browser_setting}'`
```

Comment:
Does this really temporarily change default browser? It seems too annoying for users running tests. In troubles it can leave system in different state.

What about to relax a little in here and check if default browser is Safari and if not skip the test? Would that be covered by current CI setup? Alternatively, what about to just stub the method?

Reply:
Unwritten rule is to set `ENV['CI'] == 'true'` on CIs. Would it make sense to check if browser is Safair and run test (for local usage). And force switch to Safari on CI.

Alternatively we can run this step on MacOS CI workflow, so no need to set this from within the code itself. It seems probably the best. Also it would be great to fail on CI if this is not set properly to prevent this test being silently skipped in the future.

https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables